File: a5_n^2_sort/bubble_sort.py
def bubble_sort(nums):
    for i in range(len(nums)):
        for j in range(len(nums)):
            if nums[i] < nums[j]:
                nums[i], nums[j] = nums[j], nums[i]


nums = [2, 8, 6, 4, 1]
print(nums)
bubble_sort(nums)
print(nums)

File: a5_n^2_sort/bubble_sort.cpp
#include <iostream>

using namespace std;

void bubble_sort(int n, int* arr) {
	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < n; ++j) {
			if (arr[i] < arr[j]) {
				swap(arr[i], arr[j]);
			}
		}
	}
}

int main() {
	int n;
	cin >> n;
	int* arr = new int[n];
	for (int i = 0; i < n; ++i) {
		cin >> arr[i];
	}
	bubble_sort(n, arr);
	for (int i = 0; i < n; ++i) {
		cout << arr[i] << ' ';
	}
	cout << '\n';

	delete [] arr;
	return 0;
}
